fast_matched_filter/fast_matched_filter.py

```diff
@@ -92,11 +92,30 @@ def matched_filter(templates, moveouts, weights, data, step, arch='cpu'):
         print("Compiled library for {} not loaded; exiting!".format(arch))
         return
 
-    n_templates = np.int32(templates.shape[0])
-    n_stations = np.int32(data.shape[0])
-    n_components = np.int32(data.shape[1])
+    if templates.ndim == 2:
+        n_templates = np.int32(1)
+    elif templates.ndim == 3 and data.ndim == 3:
+        n_templates = np.int32(1)
+    else:
+        n_templates = np.int32(templates.shape[0])
     n_samples_template = np.int32(templates.shape[-1])
+    
+    n_stations = np.int32(data.shape[0])
     n_samples_data = np.int32(data.shape[-1])
+    if data.ndim == 2:
+        n_components = np.int32(1)
+        data = data.reshape(n_stations, n_components, n_samples_data)
+    else:
+        n_components = np.int32(data.shape[1])
+
+    # reshape inputs if necessary
+    if templates.shape != (n_templates, n_stations, n_components, n_samples_template):
+        templates = templates.reshape(n_templates, n_stations, n_components, n_samples_template)
+    if moveouts.shape != (n_templates, n_stations, n_components):
```

Comment:
Similarly, can these if's be similar to the case above, so only what is expected will be done?